contracts/contracts/strategies/MorphoAaveStrategy.sol

```diff
@@ -0,0 +1,239 @@
+    }
+
+    /**
+     * @dev Collect accumulated rewards and send them to Harvester.
+     */
+    function collectRewardTokens()
+        external
+        override
+        onlyHarvester
+        nonReentrant
+    {
+        // Morpho Aave-v2 doesn't distribute reward tokens
+        // solhint-disable-next-line max-line-length
+        // Ref: https://developers.morpho.xyz/interact-with-morpho/get-started/interact-with-morpho/claim-rewards#morpho-aave-v2
+    }
+
+    /**
+     * @dev Get the amount of rewards pending to be collected from the protocol
+     */
+    function getPendingRewards() external view returns (uint256 balance) {
+        // Morpho Aave-v2 doesn't distribute reward tokens
+        // solhint-disable-next-line max-line-length
+        // Ref: https://developers.morpho.xyz/interact-with-morpho/get-started/interact-with-morpho/claim-rewards#morpho-aave-v2
+        return 0;
+    }
+
+    /**
+     * @dev Deposit asset into Morpho
+     * @param _asset Address of asset to deposit
+     * @param _amount Amount of asset to deposit
+     */
+    function deposit(address _asset, uint256 _amount)
+        external
+        override
+        onlyVault
+        nonReentrant
+    {
+        _deposit(_asset, _amount);
+    }
+
+    /**
+     * @dev Deposit asset into Morpho
+     * @param _asset Address of asset to deposit
+     * @param _amount Amount of asset to deposit
+     */
+    function _deposit(address _asset, uint256 _amount) internal {
+        require(_amount > 0, "Must deposit something");
+
+        IMorpho(MORPHO).supply(
+            address(_getPTokenFor(_asset)),
+            address(this), // the address of the user you want to supply on behalf of
+            _amount
+        );
+        emit Deposit(_asset, address(_getPTokenFor(_asset)), _amount);
+    }
+
+    /**
+     * @dev Deposit the entire balance of any supported asset into Morpho
+     */
+    function depositAll() external override onlyVault nonReentrant {
+        for (uint256 i = 0; i < assetsMapped.length; i++) {
+            uint256 balance = IERC20(assetsMapped[i]).balanceOf(address(this));
+            if (balance > 0) {
+                _deposit(assetsMapped[i], balance);
+            }
+        }
+    }
+
+    /**
+     * @dev Withdraw asset from Morpho
+     * @param _recipient Address to receive withdrawn asset
+     * @param _asset Address of asset to withdraw
+     * @param _amount Amount of asset to withdraw
+     */
+    function withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        _withdraw(_recipient, _asset, _amount);
+    }
+
+    function _withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) internal {
+        require(_amount > 0, "Must withdraw something");
+        require(_recipient != address(0), "Must specify recipient");
+
+        address pToken = assetToPToken[_asset];
```

Comment:
I think it would be good to use `_getPTokenFor` here to load it. This way we get a revert in case the asset is not found.

Reply:
Yep, have updated it